### PR TITLE
Bug/5164 updating mp test types

### PR DESCRIPTION
**Pull request by jordanfontainecvp**
Bug/5164 updating mp test types

### CODE REVIEW

File: src/duct-waf-workspace/duct-waf.repository.ts [hunk 1]
```diff
@@ -7,13 +7,17 @@ export class DuctWafWorkspaceRepository extends Repository<DuctWaf> {
...
+    wafEndDate: Date,
     wafBeginHour: number,
+    wafEndHour: number,
     wafValue: number,
   ): Promise<DuctWaf> {
     return this.createQueryBuilder('dw')
       .where('dw.locationId = :locationId', { locationId })
       .andWhere('dw.wafBeginDate = :wafBeginDate', { wafBeginDate })
       .andWhere('dw.wafBeginHour = :wafBeginHour', { wafBeginHour })
+      .andWhere('dw.wafEndDate = :wafEndDate', { wafEndDate })
```
Comment (billmiller13) [inline src/duct-waf-workspace/duct-waf.repository.ts]: The End Date and Hour should only be added to the where clause if EndDate (the method param) is not null.  And the resulting Where clause should look like:

If End Date is Null:
    ... AND (wafBeginDate = :wafBeginDate AND wafBeginHour = :wafBeginHour)

If End Date is not Null:
   ... AND ( (wafBeginDate = :wafBeginDate AND wafBeginHour = :wafBeginHour) OR (wafEndDate = :wafEndDate AND wafEndHour = :wafEndHour) )

File: src/monitor-qualification-workspace/monitor-qualification.repository.ts [hunk 2]
```diff
@@ -10,16 +10,18 @@ export class MonitorQualificationWorkspaceRepository extends Repository<
...
     beginDate: Date,
+    endDate: Date,
   ): Promise<MonitorQualification> {
     const result = this.createQueryBuilder('c')
       .where('c.locationId = :locId', {
         locId,
       })
       .andWhere(
-        'c.qualificationTypeCode = :qualType AND c.beginDate = :beginDate',
+        'c.qualificationTypeCode = :qualType AND c.beginDate = :beginDate AND c.endDate = :endDate',
```
Comment (billmiller13) [inline src/monitor-qualification-workspace/monitor-qualification.repository.ts]: Same fix needed as for Duct Waf

File: src/monitor-method-workspace/monitor-method.repository.ts [hunk 3]
```diff
@@ -10,12 +10,15 @@ export class MonitorMethodWorkspaceRepository extends Repository<
...
     beginDate: Date,
+    endDate: Date,
+
   ): Promise<MonitorMethod> {
     return this.findOne({
       where: {
         locationId,
         parameterCode,
         beginDate,
+        endDate,
```
Comment (billmiller13) [inline src/monitor-method-workspace/monitor-method.repository.ts]: Same fix needed as for Duct Waf.  Include endDate in where clause only if it is not null, and OR'd to the beginDate comparison

**Review by billmiller13 — CHANGES_REQUESTED** [review]
See inline comments

**Review by billmiller13 — DISMISSED** [review]
Approved with some changes I made

File: src/monitor-method-workspace/monitor-method.repository.ts [hunk 4]
```diff
@@ -10,13 +10,26 @@ export class MonitorMethodWorkspaceRepository extends Repository<
...
-        parameterCode,
-        beginDate,
-      },
-    });
+      })
+      .andWhere('mme.parameterCode = :parameterCode', {
+        parameterCode
+      })
+      .andWhere(
+        '(mm.beginDate = :beginDate AND mm.beginHour = :beginHour) OR (mm.endDate is not null AND mm.endDate = :endDate AND mm.endHour = :endHour)',
```
Comment (RobertB-CVP) [inline src/monitor-method-workspace/monitor-method.repository.ts]: Change this mm. to mme.

**Review by RobertB-CVP — CHANGES_REQUESTED** [review]

**Review by RobertB-CVP — APPROVED** [review]

**Pull request merged**